src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -181,6 +181,12 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                     _success = false;
                 }
 
+                if (!CheckPlatformVersions())
+                {
+                    // the errors will be added to the assets file
+                    _success = false;
```

Comment:
```cs
_success &= HasValidPlatformVersions();
```

Prefer positive conditions for readability. Check doesn't necessarily suggest a return boolean, but rather a side effect method.